task_sdk/tests/execution_time/conftest.py

```diff
@@ -83,7 +83,10 @@ def set_dag(what: StartupDetails, dag_id: str, task: BaseOperator) -> RuntimeTas
         task.dag = dag
         t = dag.task_dict[task.task_id]
         ti = RuntimeTaskInstance.model_construct(
-            **what.ti.model_dump(exclude_unset=True), task=t, _ti_context_from_server=what.ti_context
+            **what.ti.model_dump(exclude_unset=True),
+            task=t,
+            _ti_context_from_server=what.ti_context,
+            max_tries=what.ti_context.max_tries,
```

Comment:
I don't love that we have to mull max_tries out manually specifically. We should probably also pop it out of the server context so it isn't duplicated?
```suggestion
            max_tries=what.ti_context.popl("max_tries"),
            _ti_context_from_server=what.ti_context,

```

Reply:
Since `what.ti_context` is a Pydantic class, can't pop it

Reply:
>I don't love that we have to mull max_tries out manually specifically. 

Other option is to just have a `max_tries` as a dict value in Context instead of `ti.max_tries`